GAD2.py

```diff
@@ -0,0 +1,209 @@
+import os
+import re
+from typing import TextIO, Optional, Union
+from abc import ABC, abstractmethod
+from Bio import SeqIO, SeqUtils
+
+class BiologicalSequence(ABC):
+    @abstractmethod
+    def __len__():
+        pass
+
+    def __getitem__():
+        pass
+
+    def __str__():
+        pass
+
+    def check_sequence():
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    """Proccising nucleic acid sequences.
+    This class is parental for DNASequence and RNASequence classes.
+    """
+    def __init__(self, seq) -> None:
+        self.seq = seq
+
+    def check_sequence(self):
+        """
+        Checkes the correctnessis of input string (DNA or RNA).
+        """
+        for nucleotide in self.seq:
+            if not nucleotide in type(self).complement_rule:
+                raise ValueError (f'Incorrect nucleotide in {self.seq}')
```

Comment:
Функцию проверки лучше делать только `True/False`, а не кидать ошибку вместо `False`. Мы же делаем проверку специально, и для ошибочного сиквенса ожидаем от программы не ошибки, а именно ответа, что сиквенс неверный.
```suggestion
               return False
```